### PR TITLE
OSGI service independent approach to validate callback url

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Related Issues
- https://github.com/wso2/product-is/issues/15721

### Related PRs
- https://github.com/wso2/identity-apps/pull/3453

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -664,7 +664,7 @@
         <carbon.extension.identity.authenticator.version>3.0.0</carbon.extension.identity.authenticator.version>
         <org.wso2.carbon.identity.association.account>5.1.5</org.wso2.carbon.identity.association.account>
         <identity.extension.utils>1.0.8</identity.extension.utils>
-        <carbon.identity.framework.version>5.25.124</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.25.161-SNAPSHOT</carbon.identity.framework.version>
```
Comment (VivekVinushanth) [inline pom.xml:667]: Any reason to go with snapshot?

Comment (sadilchamishka) [reply pom.xml]: Fixed by - https://github.com/wso2/identity-apps/pull/3891/commits/828c51c15382db600153eb193bf7dc37e9feccc1

File: apps/recovery-portal/src/main/webapp/error.jsp [hunk 2]
```diff
@@ -44,13 +46,24 @@
         } else if (StringUtils.isNotBlank(request.getParameter("tenantDomain"))){
             tenantDomain = request.getParameter("tenantDomain").trim();
         }
+
+        Boolean validCallBackURL = false;
```
Comment (pavinduLakshan) [inline apps/recovery-portal/src/main/webapp/error.jsp:50]: Wouldn't it be better if we can prefix `is` to boolean variables? For example,

```suggestion
        Boolean isValidCallBackURL = false;
```

Comment (sadilchamishka) [reply apps/recovery-portal/src/main/webapp/error.jsp]: Yes. It is better. Fixed by - https://github.com/wso2/identity-apps/pull/3891/commits/c567dffe87086a9649e061686c12c755e7350637

Comment (pavinduLakshan) [reply apps/recovery-portal/src/main/webapp/error.jsp]: Thanks bro

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**